test/testdata/infer/attached_class_reopen.rb

```diff
@@ -0,0 +1,28 @@
+# typed: true
+
+class A
```

Comment:
This seems like it was a result of visiting a class def node twice for the same class in ResolveSignaturesWalk, and there was probably some sort of problem where something already existed / didn't exist unexpectedly?

In any case, could you add a comment to this test saying its intention?